cpp/include/cugraph/api_helpers.hpp

```diff
@@ -28,6 +28,14 @@ enum class sampling_strategy_t : int { UNIFORM = 0, BIASED, NODE2VEC };
 struct sampling_params_t {
   sampling_params_t(void) {}
 
+  sampling_params_t(sampling_strategy_t sampling_type,
+                    double p             = 1.0,
+                    double q             = 1.0,
+                    bool use_alpha_cache = false)
+    : sampling_type_(sampling_type), p_(p), q_(q), use_alpha_cache_(use_alpha_cache)
+  {
+  }
+
   sampling_params_t(int sampling_type, double p = 1.0, double q = 1.0, bool use_alpha_cache = false)
```

Comment:
Why are we taking `int` `sampling_type` here but `sampling_strategy_t` `sampling_type` above? Better be consistent.

And more fundamentally, why do we have two functions seemingly doing the same thing?

Reply:
The original code (developed for C++) included a constructor that takes an `int sampling_type` and converts it into the enum.  This original code is used by the current python/cython interface.

In implementing the C code, I wanted to use the enum directly (I think it's more clear in the calling code to use the enum).  I added the new constructor to allow for construction from the enum directly.  I intend to remove the old constructor once the python/cython code uses the new C API.

Reply:
Gotcha, and in that case, FIXME statement stating the intention will make this clearer.